pkg/resmgr/lib/memory/types.go

```diff
@@ -0,0 +1,318 @@
+// Copyright The NRI Plugins Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package libmem
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/containers/nri-plugins/pkg/sysfs"
+)
+
+// Type represents known types of memory.
+type Type int
+
+const (
+	TypeDRAM Type = iota // ordinary DRAM
```

Comment:
As some hardware topologies may have nodes with CPUs and without memory, should we prepare for that by having `TypeNoMem` (or something like that), too? Or do we want to handle them as DRAM or <dontcare> nodes with zero memory capacity?

Reply:
Currently we're handling them as DRAM nodes with 0 capacity.